### PR TITLE
Reservation view

### LINKED ISSUE #11: Reservation View
Create a reservation view that allows users to view information associated with a single reservation

### PR DESCRIPTION
**Pull request by jlswift2**
お願いします

### CODE REVIEW

File: reservations/models.py [hunk 1]
```diff
@@ -10,3 +10,6 @@ class Reservation(models.Model):
     end_time = models.DateTimeField()
     other_occupants = models.TextField()
     purpose = models.TextField()
+
+    def __str__(self):
+        return self.room.name
```
Comment (piebug) [inline reservations/models.py]: This is already what the room will look like as a string - so they'll effectively have the exact same name. I would suggest something like: 

```python
f"{self.user}'s reservation in {self.room}"
```

or somethin' like that

File: reservations/templates/reservations/reservation_details.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+
+<html lang-"en">
```
Comment (piebug) [inline reservations/templates/reservations/reservation_details.html]: `lang="en"` - I think you misread this from my screen share.

**Review by piebug — CHANGES_REQUESTED** [review]
do the thing

File: reservations/templates/reservations/reservation_details.html (hunk 2 above)
Comment (jlswift2) [reply reservations/templates/reservations/reservation_details.html]: Got it fixed on all html docs

File: reservations/models.py (hunk 1 above)
Comment (jlswift2) [reply reservations/models.py]: I fixed it to what you recommended. I remember reading about the f"___" thing, but I don't remember the specifics. Can you point me toward the documentation or give me a google keyword?

Comment (piebug) [reply reservations/models.py]: This is a quick run down of string formatting: https://realpython.com/python-string-formatting/. It goes over _all_ options, and `f''` strings are just one of the easier options 😄 

And here are the official docs: https://docs.python.org/3/tutorial/inputoutput.html

File: reservations/templates/reservations/reservation_details.html (hunk 2 above)
Comment (piebug) [reply reservations/templates/reservations/reservation_details.html]: ty bro

File: reservations/urls.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from django.urls import path
+
+from .views import ReservationList, ReservationDetails
+from .apps import ReservationsConfig
+
+app_name = ReservationsConfig.name
+urlpatterns = [
+    path('<int:pk>/details', ReservationDetails.as_view(), name='reservation_details'),
+    path('<int:pk>/', ReservationList.as_view(), name='reservation_list'),
```
Comment (piebug) [inline reservations/urls.py]: We need to remove the `<int:pk>` parameter here. This view is a list of ALL reservations, regardless of user or room. When you run the server, try putting different numbers in the url to see that it still brings back the exact same information.

File: reservations/views.py [hunk 4]
```diff
@@ -1,3 +1,16 @@
...
+from django.views.generic import ListView, DetailView
 
-# Create your views here.
+from .models import Reservation
+
+
+class ReservationList(ListView):
+    """Lists all the reservations associated with a room or a user"""
+    model = Reservation
+    template_name = 'reservations/reservation_list.html'
```
Comment (piebug) [inline reservations/views.py]: You're right that we will want to get lists of reservations per room and per user, but this view isn't doing that. This is ALL rooms (think of it like an admin view of everything that's going on in the system). We should change the docstring so it gives accurate info

File: seikairooms/urls.py [hunk 5]
```diff
@@ -19,5 +19,5 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('', include('rooms.urls')),
-    #path('reservation/', include('reservations.urls')),
+    path('reservation/', include('reservations.urls')),
```
Comment (piebug) [inline seikairooms/urls.py]: this is TINY and NITPICKY, but usually we use plural for these kinds of urls - so `reservations/`. Ignore `admin/` lol, it's its own thing

File: rooms/templates/rooms/room_details.html [hunk 6]
```diff
@@ -1,20 +1,21 @@
...
-<h1>Got some room details for ya!</h1>
+<h1>{{room}} Details</h1>
+<h2>Got some room details for ya!</h2>
 
 <ul>
     <li>{{room.contents}}</li>
     <li>{{room.capacity}}</li>
 </ul>
 
+<h3><a href="{% url 'reservations:reservation_list' room.pk %}">click here to see reservations associated with this room</a></li></h3>
```
Comment (piebug) [inline rooms/templates/rooms/room_details.html]: this is a lie 😏

**Comment by jlswift2** [comment]
I did not do the href to the reservation list yet. I plan to do it in another branch/PR where I simply put it up on the navbar. No real need to add it to any sort of homepage when the navbar is already there... Let me know if that is not cool.

File: rooms/templates/rooms/room_details.html (hunk 6 above)
Comment (jlswift2) [reply rooms/templates/rooms/room_details.html]: Deleted

File: seikairooms/urls.py (hunk 5 above)
Comment (jlswift2) [reply seikairooms/urls.py]: added an s and saved it in my brain.

File: reservations/views.py (hunk 4 above)
Comment (jlswift2) [reply reservations/views.py]: Corrected!

File: reservations/urls.py (hunk 3 above)
Comment (jlswift2) [reply reservations/urls.py]: Yeah. My intentions here were good but misguided. Fixed it!

File: reservations/urls.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+from django.urls import path
+
+from .views import ReservationList, ReservationDetails
+from .apps import ReservationsConfig
+
+app_name = ReservationsConfig.name
+urlpatterns = [
+    path('details', ReservationDetails.as_view(), name='reservation_details'),
```
Comment (piebug) [inline reservations/urls.py]: this path still needs the `<int:pk>` parameter at the beginning

File: reservations/templates/reservations/reservation_details.html [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{% extends 'rooms/base.html' %}
+
+{% block content %}
+
+<head>
+    <meta charset="UTF-8">
+    <title>{{reservation}} Details</title>
+</head>
+
+<body>
```
Comment (piebug) [inline reservations/templates/reservations/reservation_details.html]: The entire `<head` block and the two `<body>` tags can be removed when you extend the base

File: rooms/templates/rooms/room_details.html [hunk 9]
```diff
@@ -2,6 +2,13 @@
 
 {% block content %}
 
+<head>
+    <meta charset="UTF-8">
+    <title>{{room}} Details</title>
+</head>
+
+<body>
+
```
Comment (piebug) [inline rooms/templates/rooms/room_details.html]: no need for these!

File: rooms/urls.py [hunk 10]
```diff
@@ -5,6 +5,6 @@
 
 app_name = RoomsConfig.name
 urlpatterns = [
-    path('<int:pk>/details', RoomDetails.as_view(), name='room_details'),
+    path('<int:pk>/details/', RoomDetails.as_view(), name='room_details'),
```
Comment (piebug) [inline rooms/urls.py:8]: tyy

**Review by piebug — APPROVED** [review]
😎

**Pull request merged**